pkg/reconciler/names.go

```diff
@@ -16,10 +16,17 @@ limitations under the License.
 
 package reconciler
 
-import "fmt"
+import (
+	"fmt"
+	"github.com/knative/serving/pkg/utils"
+)
 
 func GetK8sServiceFullname(name string, namespace string) string {
-	return fmt.Sprintf("%s.%s.svc.cluster.local", name, namespace)
+	clusterDomainName, err := utils.GetClusterDomainName()
```

Comment:
Can we compute this once when the binary starts up so that we won't read a file everytime we want to get the full name of a K8s Service?

Reply:
this operation is infrequent so the should not be any quantifiable performance impact.  Also this approach allows different reaction of a failure of getting domain name. In some cases it might be useful. Appreciate other folks to chime in and share their opinion.

Reply:
If it's going to be constant, I don't see reasons for it to be recomputed each time. 
If we need to test failure cases, then go provides various methods to spoof failures (substitute files, use different file path, mockout internal function).

Reply:
ok, does it work for you, if I call it from main, set global var in utils package and then when domain name is needed utils.ClusterDomainName var can be used?

Reply:
Rather, I'd recommend to use this: https://golang.org/pkg/sync/#Once.

Reply:
Please see new solution based on once.Do()